### PR TITLE
Add root path for index files in minio

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
issue: #8214 

/kind improvement

### CODE REVIEW

File: internal/indexnode/task.go [hunk 1]
```diff
@@ -346,7 +347,8 @@ func (it *IndexBuildTask) Execute(ctx context.Context) error {
 
 		getSavePathByKey := func(key string) string {
 			// TODO: fix me, use more reasonable method
-			return strconv.Itoa(int(it.req.IndexBuildID)) + "/" + strconv.Itoa(int(it.req.Version)) + "/" + strconv.Itoa(int(partitionID)) + "/" + strconv.Itoa(int(segmentID)) + "/" + key
+			return path.Join(Params.IndexRootPath, strconv.Itoa(int(it.req.IndexBuildID))+"/"+strconv.Itoa(int(it.req.Version))+"/"+
```
Comment (xiaofan-luan) [inline internal/indexnode/task.go:350]: you should either use path.Join()
or path + "/" + path + "/"
but not combine both~

Comment (xiaocai2333) [reply internal/indexnode/task.go:350]: ok, I will fix it.

**Review by xiaofan-luan — APPROVED** [review]

File: internal/indexnode/task.go [hunk 2]
```diff
@@ -346,7 +346,8 @@ func (it *IndexBuildTask) Execute(ctx context.Context) error {
 
 		getSavePathByKey := func(key string) string {
 			// TODO: fix me, use more reasonable method
-			return strconv.Itoa(int(it.req.IndexBuildID)) + "/" + strconv.Itoa(int(it.req.Version)) + "/" + strconv.Itoa(int(partitionID)) + "/" + strconv.Itoa(int(segmentID)) + "/" + key
+			return Params.IndexRootPath + "/" + strconv.Itoa(int(it.req.IndexBuildID)) + "/" + strconv.Itoa(int(it.req.Version)) + "/" +
+				strconv.Itoa(int(partitionID)) + "/" + strconv.Itoa(int(segmentID)) + "/" + key
```
Comment (congqixia) [inline internal/indexnode/task.go:350]: Maybe `path.Join` shall be used instead of manual concatenating?

Comment (xiaocai2333) [reply internal/indexnode/task.go:350]: Use `path.Join` will nest many levels. Use ‘+’ maybe to be more intuitive？

Comment (congqixia) [reply internal/indexnode/task.go:350]: @xiaocai2333 this function signature is `func path.Join(elem ...string)`. All the elements can be joined in a single call.

Comment (xiaocai2333) [reply internal/indexnode/task.go:350]: That would be great, I will do it.

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/indexnode/param_table.go [hunk 3]
```diff
@@ -136,6 +138,14 @@ func (pt *ParamTable) initMetaRootPath() {
 	pt.MetaRootPath = path.Join(rootPath, subPath)
 }
 
+func (pt *ParamTable) initIndexRootPath() {
+	rootPath, err := pt.Load("etcd.rootPath")
```
Comment (czs007) [inline internal/indexnode/param_table.go:142]: @xiaocai2333 @xiaofan-luan  do we need a config item named minio.rootPath ?

Comment (xiaocai2333) [reply internal/indexnode/param_table.go]: 'rootPath' has been added for minio.

**Pull request merged**